ktor-utils/jvm/src/io/ktor/util/reflect/TypeInfoJvm.kt

```diff
@@ -6,30 +6,27 @@ package io.ktor.util.reflect
 
 import kotlin.reflect.*
 
+@Deprecated("Not used anymore in common code as it was needed only for JVM target.")
 public actual typealias Type = java.lang.reflect.Type
 
+/** Type with substituted generics. */
 @OptIn(ExperimentalStdlibApi::class)
-public actual inline fun <reified T> typeInfo(): TypeInfo {
-    val kType = typeOfOrNull<T>()
-    val reifiedType = try {
-        // kType != null always returns false, so we try to get javaType anyway
-        kType!!.javaType
-    } catch (_: NullPointerException) {
-        // Fallback to a type without generics if we can't get KType for the given T
-        T::class.java
-    }
-    return TypeInfo(T::class, reifiedType, kType)
-}
+public val TypeInfo.reifiedType: java.lang.reflect.Type
+    // Fallback to a type without generics if we couldn't get KType.
+    // For example, when class signature was stripped by ProGuard/R8
+    get() = kotlinType?.javaType ?: type.java
```

Comment:
Now the hack looks much better. For some reason, the problem with null check doesn't exist for a class field.